Address JNI C compilation warnings

jasonkatonica opened the pull request:
There are currently numerous JNI C compilation warnings present. This update addresses each of the warnings including the following changes:

1. Various methods needed a void argument instead of empty parentheses.

1. Prototypes were added for various methods. Specifically function pointers in use needed to have associated function pointer prototypes.

1. A mixture of unsigned long and signed long values were present. Output lengths changed are output lengths and are believed to be unsigned.

1. A few method are not in use such as `JCC_HKDF_Extract`, `JCC_HKDF_Expand`, and `JCC_HKDF` are not in use and were removed.

Signed-off-by: Jason Katonica <katonica@us.ibm.com>

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -6,7 +6,7 @@
  */
 
-typedef int (*PFI)();
+typedef int (*PFI)(void);
```
This was used for the definition of the functions later on. Do we still need it?

jasonkatonica replied:
We still seem to need these as PFI is used in the defintion of FUNC which is then used through the code. I did move this PFI defintion to a new header file named `zHardwareFunctions.h` so its now only defined in one spot although i did not try to eliminate it.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -6,7 +6,7 @@
  */
 
-typedef int (*PFI)();
+typedef int (*PFI)(void);
```
Same as before. Is this still needed?

jasonkatonica replied:
We still seem to need these as PFI is used in the defintion of FUNC which is then used through the code. I did move this PFI defintion to a new header file named `zHardwareFunctions.h` so its now only defined in one spot although i did not try to eliminate it.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -41,12 +41,41 @@ typedef struct {
-int  GCM_InitForUpdateEncrypt_core ();
-int  GCM_UpdForUpdateEncrypt_core ();
-int  GCM_FinalForUpdateEncrypt_core ();
+int GCM_InitForUpdateDecrypt_core(JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```
Would it make sense to just move the actual methods up here, instead of declaring them and implementing them later?

jasonkatonica replied:
I did move all the functions higher in the file. The empty prototypes can still be removed as those prototypes with no arguments do not represent what the method really takes as arguments. The prototypes that i created earlier are now also removed and not needed once the methods were moved.

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -38,16 +38,18 @@ typedef struct {
-char* getVersion_CCM() {
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*CCMECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*CCMGHASH_FuncPtr)(signed char *, UDATA *, signed char*, UDATA *);
```
Space before `*`.

jasonkatonica replied:
A new header file named `zHardwareFunctions.h` was created. In this file there should be a space before each `*` now including this definition.

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -38,16 +38,18 @@ typedef struct {
-
-char* getVersion_CCM() {
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*CCMECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
```
Space before `*`.

jasonkatonica replied:
A new header file named `zHardwareFunctions.h` was created. In this file there should be a space before each `*` now including this definition.

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -38,16 +38,18 @@ typedef struct {
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*CCMECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*CCMGHASH_FuncPtr)(signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*CCMzS390_FuncPtr)(unsigned char*, unsigned char*, unsigned char*, long*, long *, unsigned char*, long*);
```
Space before `*`.

jasonkatonica replied:
A new header file named `zHardwareFunctions.h` was created. In this file there should be a space before each `*` now including this definition.

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -38,16 +38,18 @@ typedef struct {
+CCMGHASH_FuncPtr CCMGHASH; // equivalent to s390_kimd_native
+CCMzS390_FuncPtr CCMzS390; // equivalent to s390_kmccm_native
+
+char* getVersion_CCM(void) {
```
Space before `*`.

jasonkatonica replied:
A space was added before this *.

KostasTsiounis commented on src/main/native/CCM.c:
```diff
@@ -458,9 +460,9 @@ JNIEXPORT jlong JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do
-                CCMECB = funcPtr[3].func;   // z_km_native_CCM
-                CCMGHASH = funcPtr[4].func; // z_kimd_native_CCM
-                CCMzS390 = funcPtr[1].func; // s390_kmccm_native
+                CCMECB = (CCMECB_FuncPtr)funcPtr[3].func;   // z_km_native_CCM
```
We should align the comment indentation.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -41,12 +41,41 @@ typedef struct {
+
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*ECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*GHASH_FuncPtr)(signed char *, UDATA *, signed char*, UDATA *);
```
Space before `*`.

jasonkatonica replied:
A new header file named zHardwareFunctions.h was created. In this file there should be a space before each * now including this definition.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -41,12 +41,41 @@ typedef struct {
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*ECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*GHASH_FuncPtr)(signed char *, UDATA *, signed char*, UDATA *);
+typedef void (*zS390_FuncPtr)(unsigned char *, unsigned char *, unsigned char *, long*, long*, unsigned char*, long*);
```
Space before `*`.

jasonkatonica replied:
A new header file named zHardwareFunctions.h was created. In this file there should be a space before each * now including this definition.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,14 +165,7 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-typedef signed char* arr;
-
-char* getVersion() {
+char* getVersion(void) {
```
Space before `*`.

jasonkatonica replied:
A space was added before this *.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -1538,7 +1560,7 @@ jbyteArray aad, jint aadLen)
 	unsigned char * key, int keyLen,
 	unsigned char * iv,  int ivLen,
-	unsigned char * aad, int aadLen,  long *updateOutlen) {
+	unsigned char * aad, int aadLen,  unsigned long *updateOutlen) {
```
I think there's 2 spaces after the comma.

jasonkatonica replied:
Changed this to be a single space.

KostasTsiounis commented on src/main/native/SymmetricCipher.c:
```diff
@@ -6,7 +6,7 @@
  */
 
-typedef int (*PFI)();
+typedef int (*PFI)(void);
```
Same as before. Is this needed?

jasonkatonica replied:
We still seem to need these as PFI is used in the defintion of FUNC which is then used through the code. I did move this PFI defintion to a new header file named zHardwareFunctions.h so its now only defined in one spot although i did not try to eliminate it.

KostasTsiounis commented on src/main/native/SymmetricCipher.c:
```diff
@@ -40,7 +40,12 @@ typedef struct OCKCipher
+typedef signed char* arr;
+
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*KMC_FuncPtr)(unsigned char*, unsigned char*, UDATA*, long, UDATA*);
```
Space before `*`.

jasonkatonica replied:
A new header file named zHardwareFunctions.h was created. In this file there should be a space before each * now including this definition.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -41,12 +41,41 @@ typedef struct {
+typedef signed char* arr;
+
+// Pointers of functions that are only available on some hardware (might be null)
+typedef void (*ECB_FuncPtr)(signed char *, signed char *, UDATA *, signed char*, UDATA *);
```
Those seem exactly the same as in `CCM`. Should we extract them into a header file or something?

jasonkatonica replied:
I consolidated the pointer defintions in the file `zHardwareFunctions.h`. In this file there are now defintions used in multiple places named `ECB_FuncPtr`, `GHASH_FuncPtr`, `zS390_FuncPtr`, and `KMC_FuncPtr`.

jasonkatonica commented:
In addition to above I updated the copyrights to have a date range where i modified the file and also added `-Werror` to platforms which are currently being built including AIX, s390x linux, and ppcle linux. This will cause failures to occur on those platforms if a warning is issued by the compiler.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
 	int           rc             = ICC_OSSL_SUCCESS;
-	static const char * functionName = "NativeInterface.GCM_decrypt_core";
+	
+	static const char * functionName = "NativeInterface.GCM_InitForUpdateDecrypt_core";
```
Should we leave a space after the `*` too?

jasonkatonica replied:
I think we should have no space after the * only before them.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+	
+#ifdef DEBUG_GCM_DETAIL
+	    if ( debug ) {
+    	    gslogMessage ("DETAIL_GCM NI GCM_InitForUpdateDecrypt_core gcmCtx %ld keyLen %d ivLen %d aadLen %d updateOutlen %ld\n", gcmCtx, keyLen, ivLen, aadLen,  *updateOutlen);
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+       
+ #ifdef DEBUG_GCM_DETAIL
+	    if ( debug ) {
+    	    gslogMessage ("DETAIL_GCM NI calling ICC_AES_GCM_iNIt\n");
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+			    /* AAD */
+#ifdef DEBUG_GCM_DETAIL
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI calling ICC_AES_GCM_DecryptUpdate for AAD\n");
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
 					ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        			if ( debug ) {			
+						gslogMessage ("NI data ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_CRYPTUPDATE_FAILED\n" );
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+					
+#ifdef DEBUG_GCM_DETAIL
+	   				if ( debug ) {
+    	    			gslogMessage ("DETAIL_GCM NI AAD ICC_AES_GCM_DecryptUpdate Succeeded\n");
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+             	
+#ifdef DEBUG_GCM_DETAIL
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI No AAD ICC_AES_GCM_DecryptUpdate not called \n");
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
 			ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_INIT_FAILED \n");
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	            gslogMessage ("DETAIL_GCM NI gslogMessage ICC_AES_GCM_NEW_FAILED\n" );
+    		}
```
Indentation seems a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+* Method:    internal functions
+* Signature: 
+*/
+int GCM_UpdForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```
Need space before `*`.

jasonkatonica replied:
I think we should have no space after the * only before them.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+/*============================================================================
+*
+* Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
+* Method:    internal functions
```
Shouldn't we provide the method name here?

jasonkatonica replied:
Updated method name in comment.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+* Signature: 
+*/
+int GCM_UpdForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+	unsigned char * data, int dataOffset, int dataLen,
```
Same as before. Should we leave a space after the `*` too?

jasonkatonica replied:
I think we should have no space after the * only before them.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-	unsigned char* iv = NULL;
-	unsigned char* aad = NULL;
-	unsigned char* parm_block = NULL;
+	static const char * functionName = "NativeInterface.GCM_UpdForUpdateDecrypt_core";
```
Same as before. Should we leave a space after the `*` too?

jasonkatonica replied:
I think we should have no space after the * only before them.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-	memcpy(parm_block + counterValueOffset, parm_block + J0Offset + blockSize - 4, 4); // Add Counter Value
-
-	zS390(input, output, aad, &len, &alen, parm_block, &mode);
+	*updateOutlen=0;
```
Spaces around `=`.

jasonkatonica replied:
Added spaces around =

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-                gslogMessage ("DETAIL_GCM rc after PrimitiveArrayCritical %d", (int) rc);
-        }
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI GCM_UpdForUpdateDecrypt_core gcmCtx %ld  dataOffset %d dataLen %d outOffset %d updateOutlen %ld\n", gcmCtx,
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+	        
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI checking for data > 0\n");
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+    		
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI ICC_AES_GCM_DecryptUpdate dataLen > 0 dataLen = %d outOffset %d updateOutlen %ld\n", 
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+			
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI data update returned ICC_AES_GCM_DecryptUpdate returns rc = %d updateOutLen = %ld\n", rc, *updateOutlen);
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+				ockCheckStatus(ockCtx);
+				return ICC_AES_GCM_CRYPTUPDATE_FAILED;
+			}
+			else {
```
Should it be `} else {`?

jasonkatonica replied:
Updated to use } else { format.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-        if ( debug ) {
-                gslogMessage ("DETAIL_GCM ret=%d", (int) ret);
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI data ICC_AES_GCM_DecryptUpdate Succeeded\n");
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+		{
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI  No data to process\n");
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+		ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+	   	if ( debug ) {
+    	    gslogMessage ("DETAIL_GCM NI  ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_NEW_FAILED\n" );
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+	}
+	return 0;
+	
+#else
```
Will this `#else` ever be exercised since we have an `#endif` a couple of lines above?

jasonkatonica replied:
Not sure if this will apply to previous block. I would assume it would not.

jasonkatonica replied:
After further inspection it seems this pattern is matched to the statement at the top of various methods such as `#if defined(AIX) || defined(WINDOWS) || defined(MAC) || defined (LINUX) || defined(__MVS__)` this else is expected to return -1 when compiled on platforms other then this.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-
+#ifdef DEBUG_GCM_DETAIL
+	   	if ( debug ) {
+    	    gslogMessage ("DETAIL_GCM NI GCM_InitForUpdateEncrypt_core updateOutlen %ld\n", *updateOutlen);
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+   	if ( debug ) {
+		gslogMessage ("NI first update rc from getOrfreeGCMCtx %d\n", rc);
+	 	gslogMessage ("DETAIL_GCM rc ICC_AES_GCM_CTX_new %d gcmCtx=%x", (int) rc, gcmCtx);
+    }
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+                   gslogMessage ("DETAIL_GCM rc ICC_AES_GCM_EncryptUpdate(aadLen > 0) %d updateAADlen %d", (int) rc, updateAADlen);
+               }
+#endif                         
+				if (rc != ICC_OSSL_SUCCESS) {
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+					ockCheckStatus(ockCtx);
+					return ICC_AES_GCM_CRYPTUPDATE_FAILED;
+				}
+				else
```
I think we should change to ` } else {`.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-					ockCheckStatus(ockCtx);
-					return ICC_AES_GCM_CRYPTFINAL_FAILED;
+			} //addLen > 0
+			else
```
I think we should change to `} else {`.

jasonkatonica replied:
Changed to } else {

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+			{
+#ifdef DEBUG_GCM_DETAIL
+        		if ( debug ) {
+					gslogMessage ("DETAIL_GCM rc  ICC_AES_GCM_EncryptUpdate not called for AAD\n");
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+#endif
+#ifdef DEBUG_GCM_DETAIL
+    if ( debug ) {
+		gslogMessage ("NI GCM_UpdForUpdateEncrypt_core updateOutlen %ld\n", *updateOutlen);
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+	//GCM Ctx cannot be null for subsequent updates
+	if (gcmCtx == 0) {
+#ifdef DEBUG_GCM_DETAIL
+        if ( debug ) {
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+           		
+#ifdef DEBUG_GCM_DETAIL
+            if ( debug ) {                  
+           		gslogMessage ("NI GCM_UpdForUpdateEncrypt_core dataLen > 0 path after ICC_AES_GCM_EncryptUpdate rc %d updateOutlen %ld\n", rc, *updateOutlen);
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+			if (rc != ICC_OSSL_SUCCESS) {
+				ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        		if ( debug ) {  				
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+		ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+        if ( debug ) {  		
+			gslogMessage ("NI ICC_AES_GCM_CTX_NEW_FAILED\n");
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+* Method:    internal functions
+* Signature: 
+*/
+int GCM_FinalForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```
Space before `*`.

jasonkatonica replied:
Should be consistant spacing now surround *.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+	    
+	    rc = ICC_OSSL_FAILURE;
+	}
+	else {
```
Switch to `} else {`.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+	if ( rc == ICC_OSSL_SUCCESS ) {
+		if (dataLen > 0) {
+#ifdef DEBUG_GCM_DETAIL
+        	if ( debug ) {  		
```
Indentation is a bit off here.

KostasTsiounis commented on src/main/native/GCM.c:
```diff
@@ -647,1562 +470,1723 @@ int GCM_encrypt_core(JNIEnv* env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+				if (rc != ICC_OSSL_SUCCESS ) {
+					// entered an error condition here
+#ifdef DEBUG_GCM_DETAIL
+        			if ( debug ) {  					
```
Indentation is a bit off here.

jasonkatonica commented on src/main/native/jgskit.mak:
```diff
@@ -32,7 +32,7 @@ AIX_LIBPATH = /usr/lib:/lib
 ifeq ($(PLATFORM),x86-linux64)
       PLAT=xa
-      CFLAGS+= -DLINUX -std=gnu99 -pedantic -Werror -Wall -fstack-protector
+      CFLAGS+= -DLINUX -Werror -std=gnu99 -pedantic -Wall -fstack-protector
```
I moved the `-Werror` flag to be consistantly after the `-DLINUX` flag. So it is in the same spot now which is why i moved this one.

KostasTsiounis replied:
Sounds good.

jasonkatonica commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
 /*============================================================================
+*
 * Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
 * Method:    internal functions
```
Updated this location and other incorrect method names through this file.

jasonkatonica replied:
Updated comment to use method name here.

jasonkatonica commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+* Signature: (JJ)V
 */
-int GCM_decrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+int GCM_InitForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
```
Change this method to use consistant spaces.

jasonkatonica commented on src/main/native/GCM.c:
```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
-	unsigned char * ciphertext, int ciphertextOffset, int ciphertextLen,
-	unsigned char * plaintext, int plaintextOffset,
-	unsigned char * aad, int aadLen, int tagLen) {
+	unsigned char * aad, int aadLen, unsigned long *updateOutlen) {
```
Change this method to use consistant spaces. I think we should have no space after the `*` only before them.

KostasTsiounis reviewed (approved):
LGTM
